### ISSUE TITLE
[4.x] The node signal preview box does not show the type of the signal parameter.

### ISSUE THREAD

**Issue by Life4gal** [godotengine/godot#65809]
### Godot version

v4.0.alpha.custom_build [d9e974cdb]

### System information

Windows 11 & OpenGL ES 3.0 & Renderer: NVIDIA GeForce RTX 3060/PCIe/SSE2

### Issue description

Since 4.0 (or since [this](https://github.com/godotengine/godot/commit/5d6e8538065050d5f5579ec03cfa9e241811e062) commit?), `GDScript` supports setting types for signal parameters, which is great, and I think setting types for parameters somehow makes the code easier to understand.

However, after reading some code about `signal`, I found that this type is not actually used, and the setting type is only supported because the parameter handling of `signal` is consistent with the normal function parameter handling.

This leads to the result shown in the picture below.
![no_signal_type](https://user-images.githubusercontent.com/52756109/190291418-a0cf6c16-1310-4dea-830b-1a55c465af4c.png)

It can be seen that the type of the parameter shows a `var`, which is too puzzling.

### Steps to reproduce

To attach a `GDScript` to any node, add a custom signal, such as:
```GDScript
signal test_signal(arg1: int, arg2, arg3: float)
```
Click on the node to view the signal preview box of the node.

### Minimal reproduction project

Not very necessary.

**Comment by Rindbee** [comment]
In fact, in e01b82857575b780dcc4788d18f128127aa8e484, the parameters of custom signals do not support typed parameters. Maybe it could be written in that way, but the parameter type doesn't resolve to the specified type. You can emit other Variant values for the parameter specified as `Vector2`.

```GDScript
extends Control

signal test(number : int)

func _ready() -> void:
	test.connect(_test)
	test.emit("test!")

func _test(arg):
	print(arg) # Prints "test!"	
```

**Comment by KoBeWi** [comment]
Looks fixed in 4.4
![image](https://github.com/user-attachments/assets/ba3b4bed-540e-42b9-9b47-f9e1c4c80369)
